evernode-governor-hook/src/governor.c

```diff
@@ -500,24 +504,15 @@ int64_t hook(uint32_t reserved)
             // ASSERT_FAILURE_MSG >> Trying to withdraw an already expired proposal.
             ASSERT(cur_ledger_timestamp - created_timestamp <= life_period);
 
-            const uint8_t *proposal_fee_ptr = &candidate_id[CANDIDATE_PROPOSAL_FEE_OFFSET];
-            const int64_t proposal_fee = INT64_FROM_BUF_LE(proposal_fee_ptr);
-
-            // Send the proposal fee to the owner.
-            etxn_reserve(1);
-            PREPARE_PAYMENT_TRUSTLINE_TX(EVR_TOKEN, issuer_accid, proposal_fee, account_field);
-            uint8_t emithash[32];
-
-            // ASSERT_FAILURE_MSG >> Emitting EVR forward txn failed
-            ASSERT(emit(SBUF(emithash), SBUF(PAYMENT_TRUSTLINE)) >= 0);
-            trace(SBUF("emit hash: "), SBUF(emithash), 1);
+            redirect_op_type = OP_REMOVE;
+        }
 
-            // Clear the proposal states.
-            // ASSERT_FAILURE_MSG >> Could not delete the candidate states.
-            ASSERT(!(state_foreign_set(0, 0, SBUF(STP_CANDIDATE_ID), FOREIGN_REF) < 0 || state_foreign_set(0, 0, SBUF(STP_CANDIDATE_OWNER), FOREIGN_REF) < 0));
+        else if (op_type == OP_REMOVE_LINKED_CANDIDATE)
+        {
+            // ASSERT_FAILURE_MSG >> Candidate removal request trx has not being initiated via registry.
+            ASSERT(BUFFER_EQUAL_20(registry_accid, account_field));
 
-            // PERMIT-MSG >> Successfully withdrawn Hook candidate proposal.
-            PERMIT();
+            redirect_op_type = OP_REMOVE;
```

Comment:
Shall we add another extra validation here by checking whether the candidate type is dud host and whether the respective host states does not exist